hts-precompile/TokenManagementContract.sol

```diff
@@ -0,0 +1,388 @@
+            name, symbol, treasury, memo, true, maxSupply, false, keys, expiry
+        );
+
+        IHederaTokenService.FixedFee[] memory fixedFees = new IHederaTokenService.FixedFee[](1);
+        fixedFees[0] = IHederaTokenService.FixedFee(1, fixedFeeTokenAddress, false, false, treasury);
+
+        IHederaTokenService.FractionalFee[] memory fractionalFees = new IHederaTokenService.FractionalFee[](1);
+        fractionalFees[0] = IHederaTokenService.FractionalFee(4, 5, 10, 30, false, treasury);
+
+        (int responseCode, address tokenAddress) =
+        HederaTokenService.createFungibleTokenWithCustomFees(token, initialTotalSupply, decimals, fixedFees, fractionalFees);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert ();
+        }
+    }
+
+    function setFreezeDefaultStatus(bool newFreezeStatus) public {
+        freezeDefaultStatus = newFreezeStatus;
+    }
+
+    function associateTokenPublic(address account, address token) public returns (int responseCode) {
+        responseCode = HederaTokenService.associateToken(account, token);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert ();
+        }
+    }
+
+    function approvePublic(address token, address spender, uint256 amount) public returns (int responseCode) {
+        responseCode = HederaTokenService.approve(token, spender, amount);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert ();
+        }
+    }
+
+    function allowancePublic(address token, address owner, address spender) public returns (int responseCode, uint256 amount) {
+        (responseCode, amount) = HederaTokenService.allowance(token, owner, spender);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert ();
+        }
+    }
+
+    function transferTokenPublic(address account, address token, int64 amount) public returns (int responseCode) {
+        IHederaTokenService.NftTransfer[] memory nftTransfers = new IHederaTokenService.NftTransfer[](0);
+
+        IHederaTokenService.AccountAmount memory accountAmountNegative =
+        IHederaTokenService.AccountAmount(msg.sender, - amount);
+        IHederaTokenService.AccountAmount memory accountAmountPositive =
+        IHederaTokenService.AccountAmount(account, amount);
+        IHederaTokenService.AccountAmount[] memory transfers = new IHederaTokenService.AccountAmount[](2);
+        transfers[0] = accountAmountNegative;
+        transfers[1] = accountAmountPositive;
+
+        IHederaTokenService.TokenTransferList memory tokenTransfer =
+        IHederaTokenService.TokenTransferList(token, transfers, nftTransfers);
+        IHederaTokenService.TokenTransferList[] memory tokenTransferList = new IHederaTokenService.TokenTransferList[](1);
+        tokenTransferList[0] = tokenTransfer;
+
+        responseCode = HederaTokenService.cryptoTransfer(tokenTransferList);
```

Comment:
Why does this use `cryptoTransfer ` instead of `transferToken`?

Reply:
This function is a wrapper for `cryptoTransfer`, which accepts only `TokenTransferList` structure as parameter. Maybe it would be better to rename this function to `cryptoTransferPublic` and add another wrapper for `transferToken` and `transferNFT` ?

Reply:
Yes, let's do the the latter suggestion.
Please also capture this same change in the relay repo.
Either yourself in a follow up PR or capture the changes in a ticket for the relay

Reply:
Okay, I have a draft PR in the relay about HTS Precompile tests, waiting for 0.30 tag for services. I can apply these changes there.